--- src/Terraria\Terraria\Chest.cs
+++ src/tModLoader\Terraria\Chest.cs
@@ -5,6 +_,8 @@
 using Terraria.GameContent.Events;
 using Terraria.ID;
 using Terraria.ObjectData;
+using Terraria.ModLoader;
+using Terraria.ModLoader.x64bit.Core;
 
 namespace Terraria
 {
@@ -167,6 +_,8 @@
 
 		public static bool isLocked(int x, int y)
 		{
+			if (Main.tile[x, y]?.type >= TileID.Count)
+				return TileLoader.IsLockedChest(x, y, Main.tile[x, y].type);
 			return Main.tile[x, y] == null || ((Main.tile[x, y].frameX >= 72 && Main.tile[x, y].frameX <= 106) || (Main.tile[x, y].frameX >= 144 && Main.tile[x, y].frameX <= 178) || (Main.tile[x, y].frameX >= 828 && Main.tile[x, y].frameX <= 1006) || (Main.tile[x, y].frameX >= 1296 && Main.tile[x, y].frameX <= 1330) || (Main.tile[x, y].frameX >= 1368 && Main.tile[x, y].frameX <= 1402) || (Main.tile[x, y].frameX >= 1440 && Main.tile[x, y].frameX <= 1474));
 		}
 
@@ -182,7 +_,7 @@
 			{
 				return item;
 			}
-			for (int i = 0; i < 1000; i++)
+			for (int i = 0; i < Core64.maxChest; i++)
 			{
 				bool flag = false;
 				bool flag2 = false;
@@ -253,6 +_,17 @@
 			int num2 = num;
 			short num3;
 			int type;
+			if (Main.tile[X, Y].type >= TileID.Count) {
+				num3 = -36;
+				type = 11;
+				bool manual = false;
+				if(!TileLoader.UnlockChest(X, Y, Main.tile[X, Y].type, ref num3, ref type, ref manual))
+					return false;
+				if (manual)
+					return true;
+				num3 *= -1;
+				goto IL_B7;
+			}
 			switch (num2)
 			{
 				case 2:
@@ -339,7 +_,7 @@
 
 		public static int FindChest(int X, int Y)
 		{
-			for (int i = 0; i < 1000; i++)
+			for (int i = 0; i < Core64.maxChest; i++)
 			{
 				if (Main.chest[i] != null && Main.chest[i].x == X && Main.chest[i].y == Y)
 				{
@@ -351,7 +_,7 @@
 
 		public static int FindChestByGuessing(int X, int Y)
 		{
-			for (int i = 0; i < 1000; i++)
+			for (int i = 0; i < Core64.maxChest; i++)
 			{
 				if (Main.chest[i] != null && Main.chest[i].x >= X && Main.chest[i].x < X + 2 && Main.chest[i].y >= Y && Main.chest[i].y < Y + 2)
 				{
@@ -364,7 +_,7 @@
 		public static int FindEmptyChest(int x, int y, int type = 21, int style = 0, int direction = 1)
 		{
 			int num = -1;
-			for (int i = 0; i < 1000; i++)
+			for (int i = 0; i < Core64.maxChest; i++)
 			{
 				Chest chest = Main.chest[i];
 				if (chest != null)
@@ -418,17 +_,25 @@
 				}
 				Main.chest[num] = chest;
 			}
-			else if (type == 21)
+			else if (type == TileID.Containers)
 			{
 				NetMessage.SendData(34, -1, -1, null, 0, (float)x, (float)y, (float)style, 0, 0, 0);
 			}
-			else if (type == 467)
+			else if (type == TileID.Containers2)
 			{
 				NetMessage.SendData(34, -1, -1, null, 4, (float)x, (float)y, (float)style, 0, 0, 0);
 			}
-			else
+			else if (type == TileID.Dressers)
 			{
 				NetMessage.SendData(34, -1, -1, null, 2, (float)x, (float)y, (float)style, 0, 0, 0);
+			}
+			else if (TileID.Sets.BasicChest[type])
+			{
+				NetMessage.SendData(34, -1, -1, null, 100, (float)x, (float)y, (float)style, 0, type, 0);
+			}
+			else if (TileLoader.IsDresser(type))
+			{
+				NetMessage.SendData(34, -1, -1, null, 102, (float)x, (float)y, (float)style, 0, type, 0);
 			}
 			return num;
 		}
@@ -460,7 +_,7 @@
 
 		public static bool CanDestroyChest(int X, int Y)
 		{
-			for (int i = 0; i < 1000; i++)
+			for (int i = 0; i < Core64.maxChest; i++)
 			{
 				Chest chest = Main.chest[i];
 				if (chest != null && chest.x == X && chest.y == Y)
@@ -480,7 +_,7 @@
 
 		public static bool DestroyChest(int X, int Y)
 		{
-			for (int i = 0; i < 1000; i++)
+			for (int i = 0; i < Core64.maxChest; i++)
 			{
 				Chest chest = Main.chest[i];
 				if (chest != null && chest.x == X && chest.y == Y)
@@ -531,12 +_,12 @@
 			}
 		}
 
-		public void AddShop(Item newItem)
+		public int AddShop(Item newItem)
 		{
 			int i = 0;
 			while (i < 39)
 			{
-				if (this.item[i] == null || this.item[i].type == 0)
+				if (this.item[i] == null || this.item[i].type == 0 || i == 38) // tModLoader change: When selling to vendor with full inventory, replace 2nd to last item rather than have the item disappear to facilitate PostSellItem Item reference.
 				{
 					this.item[i] = newItem.Clone();
 					this.item[i].favorited = false;
@@ -549,7 +_,6 @@
 					if (this.item[i].value < 1)
 					{
 						this.item[i].value = 1;
-						return;
 					}
 					break;
 				}
@@ -558,6 +_,7 @@
 					i++;
 				}
 			}
+			return i;
 		}
 
 		public static void SetupTravelShop()
@@ -882,9 +_,11 @@
 								Main.travelShop[num2++] = 3641;
 								break;
 						}
-					}
-				}
-			}
+						//patch file: num2
+					}
+				}
+			}
+			NPCLoader.SetupTravelShop(Main.travelShop, ref num2);
 		}
 
 		public void SetupShop(int type)
@@ -2229,6 +_,7 @@
 					num++;
 				}
 			}
+			NPCLoader.SetupShop(type, this, ref num);
 			if (Main.player[Main.myPlayer].discount)
 			{
 				for (int num7 = 0; num7 < num; num7++)
@@ -2243,12 +_,12 @@
 			bool[] array = new bool[1000];
 			for (int i = 0; i < 255; i++)
 			{
-				if (Main.player[i].active && Main.player[i].chest >= 0 && Main.player[i].chest < 1000)
+				if (Main.player[i].active && Main.player[i].chest >= 0 && Main.player[i].chest < Core64.maxChest)
 				{
 					array[Main.player[i].chest] = true;
 				}
 			}
-			for (int j = 0; j < 1000; j++)
+			for (int j = 0; j < Core64.maxChest; j++)
 			{
 				Chest chest = Main.chest[j];
 				if (chest != null)

