--- src/merged\Terraria.csproj
+++ src/Terraria\Terraria.csproj
@@ -132,7 +_,7 @@
     <Reference Include="System.Core" />
     <Reference Include="System.Drawing" />
     <Reference Include="Ionic.Zip.CF">
-      <HintPath>..\..\references\Ionic.Zip.CF.dll</HintPath>
+      <HintPath>Terraria.Libraries\DotNetZip\Ionic.Zip.CF.dll</HintPath>
     </Reference>
     <Reference Include="System.Windows.Forms" />
     <Reference Include="WindowsBase" />
@@ -489,7 +_,7 @@
     <Compile Include="Terraria.UI\UIState.cs" />
     <Compile Include="Terraria.UI\UserInterface.cs" />
     <Compile Include="Terraria.Utilities\CrashDump.cs" />
-    <Compile Include="Terraria.Utilities\FileOperationAPIWrapper.cs" Condition="$(DefineConstants.Contains('WINDOWS'))" />
+    <Compile Include="Terraria.Utilities\FileOperationAPIWrapper.cs" />
     <Compile Include="Terraria.Utilities\FileUtilities.cs" />
     <Compile Include="Terraria.Utilities\NPCUtils.cs" />
     <Compile Include="Terraria.Utilities\PlatformUtilities.cs" />
@@ -583,67 +_,77 @@
     <None Include="Terraria.Libraries\JSON.NET\Newtonsoft.Json.dll" />
     <EmbeddedResource Include="Terraria.Localization.Content.de-DE.Game.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.de-DE.Items.json" />
-    <EmbeddedResource Include="Terraria.Localization.Content.de-DE.json" />
+    <EmbeddedResource Include="Terraria.Localization.Content.de-DE.Main.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.de-DE.Legacy.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.de-DE.NPCs.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.de-DE.Projectiles.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.de-DE.Town.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.en-US.Game.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.en-US.Items.json" />
-    <EmbeddedResource Include="Terraria.Localization.Content.en-US.json" />
+    <EmbeddedResource Include="Terraria.Localization.Content.en-US.Main.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.en-US.Legacy.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.en-US.NPCs.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.en-US.Projectiles.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.en-US.Town.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.es-ES.Game.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.es-ES.Items.json" />
-    <EmbeddedResource Include="Terraria.Localization.Content.es-ES.json" />
+    <EmbeddedResource Include="Terraria.Localization.Content.es-ES.Main.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.es-ES.Legacy.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.es-ES.NPCs.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.es-ES.Projectiles.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.es-ES.Town.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.fr-FR.Game.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.fr-FR.Items.json" />
-    <EmbeddedResource Include="Terraria.Localization.Content.fr-FR.json" />
+    <EmbeddedResource Include="Terraria.Localization.Content.fr-FR.Main.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.fr-FR.Legacy.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.fr-FR.NPCs.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.fr-FR.Projectiles.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.fr-FR.Town.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.it-IT.Game.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.it-IT.Items.json" />
-    <EmbeddedResource Include="Terraria.Localization.Content.it-IT.json" />
+    <EmbeddedResource Include="Terraria.Localization.Content.it-IT.Main.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.it-IT.Legacy.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.it-IT.NPCs.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.it-IT.Projectiles.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.it-IT.Town.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.pl-PL.Game.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.pl-PL.Items.json" />
-    <EmbeddedResource Include="Terraria.Localization.Content.pl-PL.json" />
+    <EmbeddedResource Include="Terraria.Localization.Content.pl-PL.Main.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.pl-PL.Legacy.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.pl-PL.NPCs.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.pl-PL.Projectiles.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.pl-PL.Town.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.pt-BR.Game.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.pt-BR.Items.json" />
-    <EmbeddedResource Include="Terraria.Localization.Content.pt-BR.json" />
+    <EmbeddedResource Include="Terraria.Localization.Content.pt-BR.Main.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.pt-BR.Legacy.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.pt-BR.NPCs.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.pt-BR.Projectiles.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.pt-BR.Town.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.ru-RU.Game.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.ru-RU.Items.json" />
-    <EmbeddedResource Include="Terraria.Localization.Content.ru-RU.json" />
+    <EmbeddedResource Include="Terraria.Localization.Content.ru-RU.Main.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.ru-RU.Legacy.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.ru-RU.NPCs.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.ru-RU.Projectiles.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.ru-RU.Town.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.zh-Hans.Game.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.zh-Hans.Items.json" />
-    <EmbeddedResource Include="Terraria.Localization.Content.zh-Hans.json" />
+    <EmbeddedResource Include="Terraria.Localization.Content.zh-Hans.Main.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.zh-Hans.Legacy.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.zh-Hans.NPCs.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.zh-Hans.Projectiles.json" />
     <EmbeddedResource Include="Terraria.Localization.Content.zh-Hans.Town.json" />
   </ItemGroup>
+  <ItemGroup>
+    <Content Include="Terraria.Libraries\MojoShader.dll" />
+    <Content Include="Terraria.Libraries\SDL2.dll" />
+    <Content Include="Terraria.Libraries\SDL2_image.dll" />
+    <Content Include="Terraria.Libraries\soft_oal.dll" />
+  </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
+  <PropertyGroup>
+    <PreBuildEvent>
+    </PreBuildEvent>
+  </PropertyGroup>
 </Project>

