--- src/Terraria\Terraria.Social\SocialAPI.cs
+++ src/tModLoader\Terraria.Social\SocialAPI.cs
@@ -1,7 +_,9 @@
 using System;
 using System.Collections.Generic;
+using System.Windows.Forms;
 using Terraria.Social.Base;
 using Terraria.Social.Steam;
+using Terraria.Utilities;
 
 namespace Terraria.Social
 {
@@ -25,12 +_,22 @@
 
 		public static void Initialize(SocialMode? mode = null)
 		{
+			if (!PlatformUtilities.IsSoftwareInstalled())
+			{
+				String processorDetection = (IntPtr.Size == 8) ? "x64" : "";
+				MessageBox.Show("Buy the game on steam if you want to proceed further\n" +
+				                "If the game is owned on steam make sure your game is properly installed\n" +
+				                "Key not found in the registery!", "Terraria " + processorDetection,
+					MessageBoxButtons.OK, MessageBoxIcon.Error);
+				Environment.Exit(0);
+			}
+
 			if (!mode.HasValue)
 			{
 				mode = new SocialMode?(SocialMode.None);
 #if CLIENT
 #if DEBUG
-				if(!Program.LaunchParameters.ContainsKey("-nosteam"))
+				if (!Program.LaunchParameters.ContainsKey("-nosteam"))
 #endif
 				mode = new SocialMode?(SocialMode.Steam);
 #else

