--- src/Terraria\Terraria.Utilities\PlatformUtilities.cs
+++ src/tModLoader\Terraria.Utilities\PlatformUtilities.cs
@@ -1,7 +_,9 @@
 using SDL2;
 using System;
 using System.IO;
+using System.Linq;
 using System.Runtime.InteropServices;
+using Microsoft.Win32;
 
 namespace Terraria.Utilities
 {
@@ -41,14 +_,25 @@
 					IsLinux = true;
 					break;
 			}
+			if (IntPtr.Size == 8)
+			{
+				Is64 = true;
+			}
+			else
+			{
+				Is86 = true;
+			}
 		}
 
 		public static readonly bool IsMono = Type.GetType("Mono.Runtime") != null;
+
 		public const bool IsFNA = true;
 		public const bool IsXNA = !IsFNA;
-		public static readonly bool IsWindows = false;
 		public static readonly bool IsOSX = false;
 		public static readonly bool IsLinux = false;
+		public static readonly bool IsWindows = false;
+		public static readonly bool Is64 = false;
+		public static readonly bool Is86 = false;
 
 		public unsafe static void SavePng(Stream stream, int width, int height, int imgWidth, int imgHeight, byte[] data)
 		{
@@ -66,11 +_,57 @@
 				intPtr = intPtr2;
 			}
 			byte[] array = new byte[width * height * 4 + 41 + 57 + 256];
-			IntPtr intPtr3 = SDL.SDL_RWFromMem(array, array.Length);
+			GCHandle handle = GCHandle.Alloc(array, GCHandleType.Pinned);
+			IntPtr intPtr3 = SDL.SDL_RWFromMem(handle.AddrOfPinnedObject(), array.Length);
 			SDL_image.IMG_SavePNG_RW(intPtr, intPtr3, 1);
 			SDL.SDL_FreeSurface(intPtr);
 			int count = ((int)array[33] << 24 | (int)array[34] << 16 | (int)array[35] << 8 | (int)array[36]) + 41 + 57;
 			stream.Write(array, 0, count);
 		}
+
+		public static string ConvertToPlatformSpecificName(string name)
+		{
+			if (name == "Steamworks.NET")
+			{
+				if (IsWindows)
+				{
+					if (IntPtr.Size == 8)
+					{
+						return name + ".Windows64";
+					}
+					return name + ".Windows86";
+				}
+				if (IsOSX) { return name + ".OSX"; }
+				if (IsLinux) { return name + ".Linux"; }
+				throw new Exception("Unrecognized supported.");
+			}
+			return name;
+		}
+
+		public static bool IsCompatibleDLL(string name)
+		{
+			if (name.EndsWith(".dll")) { name = name.Substring(0, name.Length - 4); }
+			if (name.EndsWith(".Windows64")) { return !IsWindows || !Is64; }
+			if (name.EndsWith(".Windows86")) { return !IsWindows || !Is86; }
+			if (name.EndsWith(".OSX")) { return !IsOSX; }
+			if (name.EndsWith(".Linux")) { return !IsLinux; }
+			return true;
+		}
+
+
+		public static bool IsSoftwareInstalled()
+		{
+			if (PlatformUtilities.IsWindows)
+			{
+				var key = Registry.LocalMachine.OpenSubKey(@"SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Steam App 105600");
+
+				if (key == null)
+					return false;
+				else
+					return true;
+			}
+
+			return true;
+		}
 	}
 }

