--- src/Terraria\Terraria.Graphics.Capture\CaptureCamera.cs
+++ src/tModLoader\Terraria.Graphics.Capture\CaptureCamera.cs
@@ -6,7 +_,7 @@
 using System.Drawing.Imaging;
 using System.IO;
 
-#if WINDOWS
+#if XNA
 using System.Runtime.InteropServices;
 #endif
 using System.Threading;
@@ -192,7 +_,7 @@
 					{
 						for (int j = 0; j < area.Width; j++)
 						{
-#if WINDOWS
+#if XNA
 							ptr2[2] = ptr4[0];
 							ptr2[1] = ptr4[1];
 							ptr2[0] = ptr4[2];
@@ -230,7 +_,7 @@
 							"Captures",
 							Path.DirectorySeparatorChar
 						}));
-#if WINDOWS
+#if XNA
 				using (Bitmap bitmap = new Bitmap(width, height))
 				{
 					System.Drawing.Rectangle rect = new System.Drawing.Rectangle(0, 0, width, height);
@@ -277,7 +_,7 @@
 			{
 				for (int j = 0; j < width; j++)
 				{
-#if WINDOWS
+#if XNA
 						byte b = this._outputData[num + 2];
 						this._outputData[num2 + 2] = this._outputData[num];
 						this._outputData[num2] = b;
@@ -294,7 +_,7 @@
 				}
 				num += texture.Width - width << 2;
 			}
-#if WINDOWS
+#if XNA
 			using (Bitmap bitmap = new Bitmap(width, height))
 			{
 				System.Drawing.Rectangle rect = new System.Drawing.Rectangle(0, 0, width, height);

